File: app/src/main/java/com/agrogames/islandsofwar/engine/abs/unit/IUnit.java
package com.agrogames.islandsofwar.engine.abs.unit;

import com.agrogames.islandsofwar.engine.abs.common.Point;
import com.agrogames.islandsofwar.engine.abs.weapon.IWeapon;
import com.agrogames.islandsofwar.engine.abs.gamevalue.IntValue;
import com.agrogames.islandsofwar.engine.abs.map.MapObject;
import com.agrogames.islandsofwar.engine.abs.renderable.RenderableObject;
import com.agrogames.islandsofwar.engine.abs.updatable.UpdatableObject;

public interface IUnit extends MapObject, RenderableObject, UpdatableObject {
    IntValue getHealth();
    Point getLocation();
    int getMinDamage();
    IWeapon[] getWeapons();
    void loseHealth(int lost);
    void addTsd(float deltaTime);
}

File: app/src/main/java/com/agrogames/islandsofwar/engine/abs/unit/IUnit.kt
package com.agrogames.islandsofwar.engine.abs.unit

import com.agrogames.islandsofwar.engine.abs.gamevalue.IntValue
import com.agrogames.islandsofwar.engine.abs.map.MapObject
import com.agrogames.islandsofwar.engine.abs.renderable.RenderableObject
import com.agrogames.islandsofwar.engine.abs.updatable.UpdatableObject
import com.agrogames.islandsofwar.engine.abs.weapon.IWeapon

interface IUnit : MapObject, RenderableObject, UpdatableObject {
    val health: IntValue
    val minDamage: Int
    val weapons: Array<IWeapon>
    fun loseHealth(lost: Int)
    fun addTsd(deltaTime: Float)
}